File: LeetCode/Best Time to Buy and Sell Stock/best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        if (prices.count == 1) { return 0 }
        var profit : Int = 0
        var min : Int = prices[0]
        for i in 1..<prices.count {
            let current : Int = prices[i]
            if (current == min) { continue }
            if (current < min) { min = current }
            else if (profit < current - min) { profit = current - min }
        }
        return profit
    }
}

File: LeetCode/Best Time to Buy and Sell Stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 1) return 0;
        int profit = 0;
        int min = prices[0];
        for(int i = 1; i < prices.length; i++) {
            int current = prices[i];
            if (current == min) continue;
            if (current < min) min = current;
            else if (profit < current - min) profit = current - min;
        }
        return profit;
    }
}